### PR TITLE
Migrate resource ec_deployment_extension to terraform-plugin-framework

### PR DESCRIPTION
**Pull request by pascal-hofmann**
Migrate resource ec_deployment_extension to terraform-plugin-framework
## Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Refactoring (improves code quality but has no user-facing effect)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation

## Readiness Checklist

- [x] My code follows the code style of this project
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly
- [x] I have added tests to cover my changes
- [x] All new and existing tests passed

### CODE REVIEW

File: ec/ecresource/extensionresource/create.go [hunk 1]
```diff
@@ -1,73 +1,65 @@
...
-	body := extensionapi.CreateParams{
-		API:         client,
-		Name:        name,
-		Version:     version,
-		Type:        extensionType,
-		Description: description,
-		DownloadURL: downloadURL,
+	found, diags := r.read(newState.ID.Value, &newState)
+	response.Diagnostics.Append(diags...)
+	if !found {
```
Comment (dimuon) [inline ec/ecresource/extensionresource/create.go:51]: why can't we remove the resource if it's not found? I think this is how the current implementation does.

Comment (pascal-hofmann) [reply ec/ecresource/extensionresource/create.go:51]: This is not supported with the framework. It will bail out. (Same with update.)

Comment (dimuon) [reply ec/ecresource/extensionresource/create.go:51]: I think we also do something like that in the new implementation - https://github.com/dimuon/terraform-provider-ec/blob/feature/530/migrate-to-plugin-framework/ec/ecresource/trafficfilterresource/create.go#L61. 
Do I miss something?

Comment (pascal-hofmann) [reply ec/ecresource/extensionresource/create.go:51]: Good catch. I guess this needs to be changed there too.

Comment (pascal-hofmann) [reply ec/ecresource/extensionresource/create.go:51]: See https://github.com/hashicorp/terraform-plugin-framework/blob/7541ab15654b00837015180ecdb7f439e604c6cf/internal/fwserver/server_createresource.go#L113

Comment (dimuon) [reply ec/ecresource/extensionresource/create.go:51]: Can we add an error (e.g. "resource is not found") to diags and remove the state? It should be fine in that case...

Comment (pascal-hofmann) [reply ec/ecresource/extensionresource/create.go:51]: Ok, I'll try that. The same for `Update(…)`?

https://github.com/hashicorp/terraform-plugin-framework/blob/7541ab15654b00837015180ecdb7f439e604c6cf/internal/fwserver/server_updateresource.go#L125

Comment (dimuon) [reply ec/ecresource/extensionresource/create.go:51]: Yeah, why not (if it was like that before)

Comment (pascal-hofmann) [reply ec/ecresource/extensionresource/create.go:51]: Changed accordingly.

**Comment by pascal-hofmann** [comment]
@dimuon Can this be merged?

**Review by dimuon — APPROVED** [review]
LGTM

**Comment by dimuon** [comment]
@pascal-hofmann , please feel free to squash and merge it.

**Pull request merged**